src/ProjectTemplates/Microsoft.Extensions.AI.Templates/src/ChatWithCustomData/ChatWithCustomData-CSharp.AppHost/Program.cs

```diff
@@ -0,0 +1,77 @@
+var builder = DistributedApplication.CreateBuilder(args);
+#if (IsOllama) // ASPIRE PARAMETERS
+#elif (IsGHModels)
+
+// You will need to set the endpoint and key to your own values
+// You can do this using Visual Studio's "Manage User Secrets" UI, or on the command line:
+//   cd this-project-directory
+//   dotnet user-secrets set Parameters:gitHubModelsToken YOUR-GITHUB-TOKEN
+var gitHubModelsToken = builder.AddParameter("gitHubModelsToken", secret: true);
+#else // IsAzureOpenAI || IsOpenAI
+
+// You will need to set the connection string to your own value
+// You can do this using Visual Studio's "Manage User Secrets" UI, or on the command line:
+//   cd this-project-directory
+#if (IsOpenAI)
+//   dotnet user-secrets set ConnectionStrings:openai Key=YOUR-API-KEY
+#else // IsAzureOpenAI
+//   dotnet user-secrets set ConnectionStrings:openai Endpoint=https://YOUR-DEPLOYMENT-NAME.openai.azure.com;Key=YOUR-API-KEY
+#endif
+var openai = builder.AddConnectionString("openai");
+#endif
+#if (UseAzureAISearch)
+
+// You will need to set the connection string to your own value
+// You can do this using Visual Studio's "Manage User Secrets" UI, or on the command line:
+//   cd this-project-directory
+//   dotnet user-secrets set ConnectionStrings:azureAISearch Endpoint=https://YOUR-DEPLOYMENT-NAME.search.windows.net;Key=YOUR-API-KEY
+var azureAISearch = builder.AddConnectionString("azureAISearch");
+#elif (UseQdrant)
+
+var qdrantApiKey = builder.AddParameter("qdrantApiKey", secret: true);
+#else // UseLocalVectorStore
+#endif
+#if (IsOllama) // AI SERVICE PROVIDER CONFIGURATION
+
+var ollama = builder.AddOllama("ollama")
+    .WithDataVolume();
+var chat = ollama.AddModel("chat", "llama3.2");
+var embeddings = ollama.AddModel("embeddings", "all-minilm");
+#endif
+#if (UseAzureAISearch) // VECTOR DATABASE CONFIGURATION
+#elif (UseQdrant)
+
+var vectorDB = builder.AddQdrant("vectordb", apiKey: qdrantApiKey, grpcPort: 6334, httpPort: 6333)
+    .WithDataBindMount("./qdrant_data")
```

Comment:
Could you clarify why we'd mount `qdrant_data` inside the repo source code? Could we use `.WithDataVolume()` instead, so the developer doesn't need to exclude `qdrant_data` from source control?

Reply:
Yes, we could probably use `.WithDataVolume()`. I was just experimenting with this locally yesterday but hit other blocking issues that prevented me from validating it at the time. I'll see if we can make this change.

We could probably not require specifying an API key as well.